File: a008.py
# import
import sys


# function area
def factorSum (n):
	summation = 0
	if n % 2 == 0:
		ranger = n / 2
	else:
		ranger = int (n / 3)
	for i in range (ranger):
		if n % (i + 1) == 0:
			summation += (i + 1)
	return summation




# driver code
for line in sys.stdin:
	n = eval (line)
	if (n == 0) :
		break
	else :
		SUM = factorSum (n)
		if (n == SUM):
			print ("={:d}".format (n))
		elif  (n == factorSum (SUM)):
			print ("{:d}".format (SUM))
		else:
			print ("0")

File: a008.c
#include <stdio.h>
#include <stdlib.h>

int factorSum (int n)
{
	int range, sum = 0;
	if (n % 2 == 0)
		range = n / 2;
	else
		range = n / 3;
	for (int i = 1; i <=range; i++)
		if (n % i == 0)
			sum += i;
	return sum;
}



int main ()
{
	int input;
	while (scanf ("%d", &input) != EOF) {
		if (input == 0) break;

		int SUM = factorSum (input);
		if (input == SUM)
			printf ("=%d\n", input);
		else if (input == factorSum (SUM))
			printf ("%d\n", SUM);
		else
			printf ("0\n");
	}
}
